src/hacbs/components/ApplicationDetails/tabs/overview/visualization/utils/__tests__/visualization-utils.spec.ts

```diff
@@ -130,60 +132,51 @@ describe('workflowToNodes', () => {
   });
 });
 
-describe('sortEnvironments', () => {
-  test('should return empty array for invalid values', () => {
-    expect(sortEnvironments(null)).toHaveLength(0);
-    expect(sortEnvironments(undefined)).toHaveLength(0);
-    expect(sortEnvironments([])).toHaveLength(0);
+describe('getLastEnvironments', () => {
+  test('should return default environment for invalid values', () => {
+    expect(getLastEnvironments([])).toEqual(['static-env']);
+    expect(getLastEnvironments(null)).toEqual(['static-env']);
+    expect(getLastEnvironments(undefined)).toEqual(['static-env']);
   });
 
-  test('should return sorted array of environments', () => {
-    const sortedEnvironments = sortEnvironments(sampleEnvironments);
-    expect(sortedEnvironments).toHaveLength(3);
-    expect(sortedEnvironments[0]).toBe('test-environment');
-    expect(sortedEnvironments[1]).toBe('staging-environment');
-    expect(sortedEnvironments[2]).toBe('production-environment');
+  test('should return the last environments in the list', () => {
+    expect(getLastEnvironments(sampleEnvironments)).toEqual(['production-environment']);
   });
-});
 
-describe('getLastEnvironment', () => {
-  test('should return default environment for invalid values', () => {
-    expect(getLastEnvironment([])).toBe('static-env');
-    expect(getLastEnvironment(null)).toBe('static-env');
-    expect(getLastEnvironment(undefined)).toBe('static-env');
+  test('should contain the environments with non existing parent name', () => {
+    expect(getLastEnvironments(sampleEnvironmentsWithInvalidParentEnvironment)).toEqual([
+      'production-environment',
+      'testing-environment',
+    ]);
   });
+});
 
-  test('should return the last environment in the list', () => {
-    expect(getLastEnvironment(sampleEnvironments)).toBe('production-environment');
+describe('getMaxName', () => {
```

Comment:
you can merge the getmaxName positive and negative checks into one

Reply:
done.